src/app/app.component.html

```diff
@@ -0,0 +1,54 @@
+<div class="mat-app-background">
+  <div class="header">
+    <mat-toolbar class="toolbar" color="primary">
+      <h1 class="toolbar__title">{{ title }}</h1>
+      <div class="toolbar__options">
+        <app-filter (filterChange)="onFilterChange($event)"></app-filter>
+      </div>
+    </mat-toolbar>
+    <mat-progress-bar *ngIf="loading" class="progress-bar" mode="indeterminate"></mat-progress-bar>
+  </div>
+
+  <table class="table" mat-table matSort [dataSource]="dataSource">
+    <ng-container *ngFor="let column of displayedColumns" [matColumnDef]="column">
+      <th
+        mat-header-cell
+        *matHeaderCellDef
+        mat-sort-header
+        [sortActionDescription]="'Sort by ' + column"
+      >
+        {{ column.toUpperCase() }}
+      </th>
+      <div *ngIf="column !== 'type'">
+        <td mat-cell *matCellDef="let element">{{ element[column] }}</td>
+      </div>
+      <div *ngIf="column === 'type'">
+        <td mat-cell *matCellDef="let element">
+          <div
+            [ngClass]="{
+              dot: true,
+              'dot--red': typeColors[element[column]].color === 'red',
+              'dot--orange': typeColors[element[column]].color === 'orange',
+              'dot--yellow': typeColors[element[column]].color === 'yellow',
+              'dot--blue': typeColors[element[column]].color === 'blue',
+              'dot--grey': typeColors[element[column]].color === 'grey'
+            }"
```

Comment:
not bad but, can we do something like`'dot--' + typeColors[element[column]].color` ?

Reply:
That's not possible, but you made me think of a different way of defining ngClass which enables your solution.